pkg/apis/pipeline/v1alpha1/pull_request_resource.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2018 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/tektoncd/pipeline/pkg/names"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	prSource = "pr-source"
+	// The container that we use to implement the PR source step.
+	prImage = flag.String("pr-image", "override-with-pr:latest",
+		"The container image containing our PR binary.")
+)
+
+// PullRequestResource is an endpoint from which to get data which is required
+// by a Build/Task for context.
+type PullRequestResource struct {
+	Name             string               `json:"name"`
+	Type             PipelineResourceType `json:"type"`
+	URL              string               `json:"url"`
+	GithubOauthToken string               `json:"githubOauthToken"`
+	//Secrets holds a struct to indicate a field name and corresponding secret name to populate it
+	Secrets []SecretParam `json:"secrets"`
+}
+
+// NewPullRequestResource create a new git resource to pass to a Task
+func NewPullRequestResource(r *PipelineResource) (*PullRequestResource, error) {
+	if r.Spec.Type != PipelineResourceTypePullRequest {
+		return nil, fmt.Errorf("PipelineResource: Cannot create a PR resource from a %s Pipeline Resource", r.Spec.Type)
+	}
+	prResource := PullRequestResource{
+		Name:    r.Name,
+		Type:    r.Spec.Type,
+		Secrets: r.Spec.SecretParams,
+	}
+	for _, param := range r.Spec.Params {
+		switch {
+		case strings.EqualFold(param.Name, "URL"):
+			prResource.URL = param.Value
+		}
+	}
+
+	return &prResource, nil
+}
+
+// GetName returns the name of the resource
+func (s PullRequestResource) GetName() string {
+	return s.Name
+}
+
+// GetType returns the type of the resource, in this case "Git"
+func (s PullRequestResource) GetType() PipelineResourceType {
+	return PipelineResourceTypePullRequest
+}
+
+// GetURL returns the url to be used with this resource
+func (s *PullRequestResource) GetURL() string {
+	return s.URL
+}
+
+// GetParams returns the resource params
+func (s PullRequestResource) GetParams() []Param { return []Param{} }
```

Comment:
im confused about why `GetParams` exists when every instance of this interface returns an empty list :thinking:

Reply:
:thinking:

Reply:
yep, @wlynch did you create an issue already ?

Reply:
https://github.com/tektoncd/pipeline/issues/991